src/App.tsx

```diff
@@ -15,37 +19,92 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabetically = 'Alphabetically',
+  Length = 'Length',
+}
+
+enum ButtonType {
+  Alphabetically = 'Sort alphabetically',
+  Length = 'Sort by length',
+  Reverse = 'Reverse',
+  Reset = 'Reset',
+}
+
+function getSortedGoods(
+  goods: string[],
+  sortField: SortType | '',
+  reverse: boolean,
+): string[] {
+  let sortedGoods = [...goods];
+
+  switch (sortField) {
+    case SortType.Alphabetically:
+      sortedGoods.sort();
+      break;
+
+    case SortType.Length:
+      sortedGoods.sort((a, b) => a.length - b.length);
+      break;
+
+    default:
+      break;
+  }
+
+  if (reverse) {
+    sortedGoods = sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sort, setSort] = useState<SortType | ''>('');
+  const [reverse, setReverse] = useState<boolean>(false);
```

Comment:
It is ok just to 
```suggestion
  const [reverse, setReverse] = useState(false);
```

TypeScript can guess the type